packages/lesswrong/server/callbacks/commentCallbacks.ts

```diff
@@ -283,24 +270,33 @@ async function sendModerationPM({ firstMessageContents, lwAccount, comment, noEm
 }
 
 async function commentsRejectSendPMAsync (comment: DbComment, currentUser: DbUser) {
-  const onWhat = comment.tagId
-    ? (await Tags.findOne(comment.tagId))?.name
-    : (comment.postId
-      ? (await Posts.findOne(comment.postId))?.title
-      : null
-    );
+  let rejectedContentLink = "[Error: content not found]"
+  let contentTitle: string|null = null
+
+  if (comment.tagId) {
+    const tag = await Tags.findOne(comment.tagId)
+    if (tag) {
+      contentTitle = tag.name
+      rejectedContentLink = `<a href="https://lesswrong.com/tag/${tag.slug}/discussion?commentId=${comment._id}">comment on ${tag.name}</a>`
+    }
+  } else if (comment.postId) {
+    const post = await Posts.findOne(comment.postId)
+    if (post) {
+      contentTitle = post.title
+      rejectedContentLink = `<a href="https://lesswrong.com/posts/${post._id}/${post.slug}?commentId=${comment._id}">comment on ${post.title}</a>`
+    }
+  }
 
   const commentUser = await Users.findOne({_id: comment.userId})
 
-  let firstMessageContents =
+  let messageContents =
       // TODO: make link conditional on forum, or something
-      `Unfortunately, I rejected your comment on "${onWhat}".  (The LessWrong moderator team is raising its moderation standards, see <a href="https://www.lesswrong.com/posts/kyDsgQGHoLkXz6vKL/lw-team-is-adjusting-moderation-policy">this announcement</a> for details).`
+      `Unfortunately, I rejected your ${rejectedContentLink}.  (The LessWrong moderator team is raising its moderation standards, see <a href="https://www.lesswrong.com/posts/kyDsgQGHoLkXz6vKL/lw-team-is-adjusting-moderation-policy">this announcement</a> for details).`
 
   if (comment.rejectedReason) {
-    firstMessageContents += ` Your post didn't meet the bar for at least the following reason(s): ${comment.rejectedReason}`;
+    messageContents += ` <p>Your post didn't meet the bar for at least the following reason(s):</p><p>${comment.rejectedReason}</p>`;
```

Comment:
I realize that this was already wrong before your PR, but probably just a good time to fix `post` to `comment`.